### ISSUE TITLE
Use Client in an async fashion

### ISSUE THREAD

**Issue by dwhiteddsoft** [dapr/rust-sdk#52]
I simply want to create a client and pass it to a number of threads to do a publish_event (just looking at throughput). No matter how I try to move, clone, copy Client into another thread, it fails in various ways. Any thoughts on how to accomplish? Here is some example code you can beat up:
```rust
    // Create the client
    let mut client = dapr::Client::<dapr::client::TonicClient>::connect(addr).await.unwrap();
    let foo = std::sync::Arc::new(client);
    for x in 0..10 {
        let cli = foo.clone();
        tokio::spawn(async move {
            println!("In spawn");
            let mut metadata = HashMap::<String, String>::new();
            metadata.insert("count".to_string(), x.to_string());
            cli
                .publish_event(
                    "pubsub".to_string(),
                    "A".to_string(),
                    "text/plain".to_string(),
                    format!("{} => hello from rust!", &x).into_bytes(),
                    Some(metadata),
                )
                .await;
        });
    }
```

**Comment by dwhiteddsoft** [comment]
FYI here is the full code sample that works on an R-Pi4. The problem is this is SLOOOOOWWWWWW. This gets through messages at about 191 messages/sec. On an example C# multi-thread program I can get 754 messages/sec. I imagine a bulk of the time is spent in creating and tearing down the client.
```rust
use std::{collections::HashMap, thread, time::{Duration, Instant}};

#[tokio::main]
async fn main() {
    // TODO: Handle this issue in the sdk
    // Introduce delay so that dapr grpc port is assigned before app tries to connect
    thread::sleep(Duration::from_secs(2));
    let start = Instant::now();
    let mut handles = vec![];
    for x in 0..1000 {
        handles.push(sender(x));
    }
    futures::future::join_all(handles).await;
    println!("All done. Elapsed: {} ms", start.elapsed().as_secs_f32());
}
async fn sender(count: i32) -> Result<(), Box<dyn std::error::Error>> {
    //println!("--> function");
    // Get the Dapr port and create a connection
    let port: u16 = 45045; //std::env::var("DAPR_GRPC_PORT")?.parse()?;
    let addr = format!("https://127.0.0.1:{}", port);

    // Create the client
    let mut client = dapr::Client::<dapr::client::TonicClient>::connect(addr).await?;

    // name of the pubsub component
    let pubsub_name = "pubsub".to_string();

    // content type of the pubsub data
    let data_content_type = "text/plain".to_string();

    // topic to publish message to
    let topic = "A".to_string();

    // message metadata
    let mut metadata = HashMap::<String, String>::new();
    metadata.insert("count".to_string(), count.to_string());

    // message
    let message = format!("{} => hello from rust!", &count).into_bytes();

    client
        .publish_event(
            &pubsub_name,
            &topic,
            &data_content_type,
            message,
            Some(metadata),
        )
        .await?;
    //println!("<-- function");
    Ok(())
}
```

**Comment by wcs1only** [comment]
So the publish_event function is fully asyncronous and returns a future that can be await'd on. In general the pattern goes:
Create a batch of messages, and save their futures to a stream. Then you can await the stream of futures. I'll take a pass at the example you gave a little later today, but posting this now with the hopes that my reply might be enough to unblock you.

**Comment by mikeee** [comment]
Closing in favour of #91 as this will partially resolve this.
